docs/defi/chain-key-tokens/ckbtc/using-ckbtc-in-dapps.mdx

```diff
@@ -10,172 +10,89 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+- Rust: [icrc-ledger-agent](https://crates.io/crates/icrc-ledger-agent)
+- TypeScript: [@dfinity/ledger-icrc](https://www.npmjs.com/package/@dfinity/ledger-icrc)
+
+In this model, all transactions are executed from the user’s own [account](/docs/defi/token-standards/#accounts-principals-and-subaccounts) (typically the principal without a subaccount).
+
+### Inter-canister calls
+A canister can also interact with the ckBTC ledger.
+
+- In some cases, the canister may use ckBTC for its own purposes, such as making payments to other canisters, managing a treasury, or executing automated transfers.
+- In other cases, the canister may act on behalf of users, which requires managing user funds. This can be achieved by creating **subaccounts per user** or by **pooling all funds in a single account** and maintaining an internal balance table.
+  - It is recommended to use the [ICRC-2](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-2/README.md) approval flow (`icrc2_approve` / `icrc2_transfer_from`) to move funds from a user’s wallet to the backend canister.
+
+Libraries are available to simplify inter-canister calls:
+
+- Rust: [icrc-ledger-client-cdk](https://crates.io/crates/icrc-ledger-client-cdk)
+- Motoko: [icrc2-types](https://mops.one/icrc2-types) (includes ICRC-1 types)
+
+Example code:
+- Motoko: [transfer](https://github.com/dfinity/examples/tree/master/motoko/token_transfer), [transfer_from](https://github.com/dfinity/examples/tree/master/motoko/token_transfer_from)
+- Rust: [transfer](https://github.com/dfinity/examples/tree/master/rust/token_transfer), [transfer_from](https://github.com/dfinity/examples/tree/master/rust/token_transfer_from)
+
+## Bitcoin ↔ ckBTC conversion
+
+Beyond ledger interactions, a dapp may also support minting and burning ckBTC by calling the [ckBTC minter canister](https://dashboard.internetcomputer.org/canister/mqygn-kiaaa-aaaar-qaadq-cai).
+
+This can be done in two ways:
+
+- **Client calls (via agent)**
+  - Any client can call the ckBTC minter canister directly.
+  - TypeScript library: [@dfinity/ckbtc](https://www.npmjs.com/package/@dfinity/ckbtc)
+
+- **Inter-canister calls**
+  - A backend canister can call the ckBTC minter canister as part of its logic, either for its own use or on behalf of users.
+
+Check out the [ckBTC API](/docs/references/ckbtc-reference) and learn more about implementing this conversion flow in the [ckBTC best practices](/docs/defi/defi-best-practices).
+
+### Why integrate conversion into a dapp
+- Enables deposits from Bitcoin → ckBTC and withdrawals from ckBTC → Bitcoin without leaving the application.
+- Reduces friction and improves overall user experience.
+
+**Note:** Applications that integrate with [OISY](https://oisy.com) may rely on OISY to handle Bitcoin ↔ ckBTC conversion on behalf of their users, instead of implementing the conversion themselves.
+
+## Examples applications for ckBTC
 
 - [Bitcoin point of sale app](/docs/references/samples/motoko/ic-pos/).
```

Comment:
if there is only one example we can show here, should we remove this section? looks a bit empty otherwise

Reply:
I might find another good one. let me see.